### PR TITLE
Use basic authentication credentials if found in the URL

### LINKED ISSUE #105: Auth for sites with authentication
Support for authentication, at least basic would be great.

Support for github auth would be great too.

### PR DESCRIPTION
**Pull request by woneill**
Close #105

---

*edited*

### CODE REVIEW

**Comment by dkhamsing** [comment]
@woneill thx!

File: README.md [hunk 1]
```diff
@@ -86,6 +86,9 @@ $ awesome_bot README.md --allow 403,429
 
 $ awesome_bot README.md --base-url https://github.com/IDR/idr-notebooks/blob/master/
 # will check relative links using the base url provided
+
+$ awesome_bot README.md --base-url https://username:accesstoken@github.com/Organization/Private-repo/blob/master/
```
Comment (dkhamsing) [inline README.md]: this example is the confusing part.. it suggests username:password only works with --base-url

hmm

Comment (woneill) [inline README.md]: Yeah, wasn't really sure where/how to put it in the README. Do you think it's better to take it out altogether and let people find out organically that it works?

Comment (dkhamsing) [inline README.md]: maybe, it doesn't look like too many people are asking for this

yeah let's remove this from the readme

**Comment by dkhamsing** [comment]
hmm some tests would be great

do u know of a site with basic auth that people can use for testing??

**Comment by dkhamsing** [comment]
ah we can use http://httpbin.org/basic-auth/user/passwd

**Comment by woneill** [comment]
Good find! I'll work up some tests using that

**Comment by dkhamsing** [comment]
ya `http://user/passwd:httpbin.org/basic-auth/user/passwd` :)

**Comment by dkhamsing** [comment]
> http://user:passwd@httpbin.org/basic-auth/user/passwd

**Comment by dkhamsing** [comment]
nice test buddy!

File: spec/check_spec.rb [hunk 2]
```diff
@@ -57,6 +57,30 @@
       end
     end
 
+    context "given 2 links with basic-auth embedded" do
```
Comment (dkhamsing) [inline spec/check_spec.rb]: do you mind moving these to `spec/status_spec.rb` :)

**Comment by woneill** [comment]
Gah- apologies for not taking care of https://github.com/dkhamsing/awesome_bot/pull/134#discussion_r119714101 myself! Got pulled into another issue and hadn't had a chance to followup. Poor communication on my part!

Thanks for taking care of it for me!

**Comment by dkhamsing** [comment]
np :)

**Pull request merged**